src/core/runtime.cpp

```diff
@@ -2673,6 +2674,8 @@ void SHWire::cleanup(bool force) {
       }
     }
 
+    dispatcher.trigger(SHWire::OnStopEvent{this});
```

Comment:
This is not correct, the event is on stop but happens on cleanup

Reply:
Yeah I guess I can add a second one, I needed a correct place to clean up variables I manually inserted into the wire

Reply:
Add an on cleanup event I guess

Reply:
Also I know the issue, stop often doesn't get called at all!!

Reply:
For now add on cleanup new event. I'll fix the stop thing.

Reply:
Fixed in Tracy PR, can remove this. Hopefully works for you.

Reply:
Added a dedicated OnCleanup event for this